pkg/apis/core/types.go

```diff
@@ -3496,9 +3499,21 @@ type PodStatus struct {
 	// give the resources on this node to a higher priority pod that is created after preemption.
 	// +optional
 	NominatedNodeName string
+
+	// HostIP holds the IP address of the host to which the pod is assigned. Empty if the pod has not started yet.
+	// A pod can be assigned to a node that has a problem in kubelet which in turns mean that HostIP will
+	// not be updated even if there is a node is assigned to pod
 	// +optional
 	HostIP string
 
+	// HostIPs holds the IP addresses allocated to the host. If this field is specified, the first entry must
+	// match the hostIP field. This list is empty if the pod has not started yet.
+	// A pod can be assigned to a node that has a problem in kubelet which in turns means that HostIPs will
+	// not be updated even if there is a node is assigned to this pod.
+	// match the hostIP field. This list is empty if no IPs have been allocated yet.
+	// +optional
+	HostIPs []HostIP
```

Comment:
@wzshiming Do you recall why we decided to add a field to internal version, rather than do it the same way we did PodIPs?  That puts sopme code into conversion but is easier overall to manage, I think...

Reply:
As far as I can remember, there was a time when `PodIP` and `PodIPs` coexisted in the internal version, but then `PodIP` was gradually replaced, and finally deleted.

Reply:
I asked this here https://github.com/kubernetes/kubernetes/pull/109616#discussion_r858374927
I'm almost sure we replaced PodIP directly

Reply:
trying to page in all the history around that change, let's make sure whatever we do applies the lessons learned in:
* https://github.com/kubernetes/kubernetes/pull/73977
* https://github.com/kubernetes/kubernetes/issues/88494
  * https://github.com/kubernetes/kubernetes/pull/88505
* https://github.com/kubernetes/kubernetes/pull/88988
* https://github.com/kubernetes/kubernetes/issues/89888
  * https://github.com/kubernetes/kubernetes/pull/89914
* https://github.com/kubernetes/kubernetes/issues/90625
  * https://github.com/kubernetes/kubernetes/pull/90628

Reply:
I reviewed these changes and only #88505 was not included, as this `conversion` function was removed from this work outright after learning the lesson from [#101566 commits](https://github.com/kubernetes/kubernetes/pull/101566/commits/8e347718e3b91c9f1cd2bcb83ed70c8c6ea434a3) about overloading `apiserver` due to repeatedly erasing and restoring `hostIPs` due to `conversion` in a cluster with version-skewed `apiserver` and `kubelet`.

Reply:
Is there anything else I can do?